pkg/backup/item_backupper.go

```diff
@@ -318,21 +308,57 @@ func (ib *itemBackupper) executeActions(
 	groupResource schema.GroupResource,
 	name, namespace string,
 	metadata metav1.Object,
+	finalize bool,
 ) (runtime.Unstructured, error) {
 	for _, action := range ib.backupRequest.ResolvedActions {
 		if !action.ShouldUse(groupResource, namespace, metadata, log) {
 			continue
 		}
 		log.Info("Executing custom action")
 
-		// Note: we're ignoring the operationID returned from Execute for now, it will be used
-		// with the async plugin action implementation
-		updatedItem, additionalItemIdentifiers, _, err := action.Execute(obj, ib.backupRequest.Backup)
+		updatedItem, additionalItemIdentifiers, operationID, itemsToUpdate, err := action.Execute(obj, ib.backupRequest.Backup)
 		if err != nil {
 			return nil, errors.Wrapf(err, "error executing custom action (groupResource=%s, namespace=%s, name=%s)", groupResource.String(), namespace, name)
 		}
+
 		u := &unstructured.Unstructured{Object: updatedItem.UnstructuredContent()}
 		mustInclude := u.GetAnnotations()[mustIncludeAdditionalItemAnnotation] == "true"
+		// remove the annotation as it's for communication between BIA and velero server,
+		// we don't want the resource be restored with this annotation.
+		if _, ok := u.GetAnnotations()[mustIncludeAdditionalItemAnnotation]; ok {
+			delete(u.GetAnnotations(), mustIncludeAdditionalItemAnnotation)
+		}
+		obj = u
+		if finalize {
```

Comment:
So that means when calling BIA in the `Finalization` phase, the async operation may be triggered b/c we can't control that, and we choose not to track the progress of these operations, IMO this is a little problematic. 
This probably will not happen currently, but we should at least document it in the design clearly.

Additionally, it may not be right to ignore the additional Items returned by the BIA.

Reply:
We definitely want to document for users that they should not be starting async actions if backup phase is finalize -- and I'll update my example plugin to check phase as an example. There's a real risk of an infinite loop scenario if we track async operations started by plugins during finalize, as the plugin would have to go back to waiting for plugin operations, then finalize again, where new async operations could start, etc.

Reply:
Additional items is less complicated. We can definitely process them here if we need to. In the original design, it was unnecessary since we were actually backing up these items on the first pass, then finalize was just backing them up again for updates, so the additional items were already included. But with the current implementation, these items are deferred until finalize, so they don't have the opportunity to add additional items anymore. We may want to add that back, although it will add some complications to processing. I'll consider this problem in the follow-on PR.